src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Completion/RazorCompletionEndpoint.cs

```diff
@@ -129,11 +129,18 @@ public async Task<CompletionList> Handle(CompletionParams request, CancellationT
             var tagHelperDocumentContext = codeDocument.GetTagHelperContext();
 
             var sourceText = await document.GetTextAsync();
-            var linePosition = new LinePosition(request.Position.Line, request.Position.Character);
-            var hostDocumentIndex = sourceText.Lines.GetPosition(linePosition);
+            var hostDocumentIndex = request.Position.GetAbsoluteIndex(sourceText);
             var location = new SourceSpan(hostDocumentIndex, 0);
+            var reason = request.Context.TriggerKind switch
```

Comment:
Would it be an option to re-use the existing `CompletionTriggerKind` type instead of creating the `CompletionReason` class? I understand it'd be more cases to check at various places but I feel like adding a completely new class may be a bit confusing and would create more overhead to maintain in the future.

Reply:
> I think if it were something more like if (context.TriggerKind == TriggerKind.Explicit) where in our completion endpoint we translated an isIncomplete to an equivalent of a ctrl + space then it wouldn't be as leaky

@NTaylorMullen suggested that we avoid leaking LSP knowledge into the Completion service, and I agreed that was probably for the best.

Reply:
Ah gotcha, not wanting to leak is an understandable reason. sgtm!